messenger/http_transporter.go

```diff
@@ -490,7 +490,10 @@ func (t *HTTPTransporter) messageDecoder(w http.ResponseWriter, r *http.Request)
 	// libprocess will ignore responses, see ignore_data).
 	// see https://github.com/apache/mesos/blob/adecbfa6a216815bd7dc7d26e721c4c87e465c30/3rdparty/libprocess/src/process.cpp#L2192
 	for {
-		if r, ok := <-decoder.Requests(); ok {
+		if r, ok := <-decoder.Requests(); !ok {
```

Comment:
nice catch. when there's a 'return' as a result of an if, we're trying to stick with a style like this going forward:
```go
if something {
  return
}
// do something else
```
this means declaring the r, ok outside of the if statement, and that's OK